### PR TITLE
Draft for source to source transformation

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

File: content/2019-03-06-source-to-source-transformation.rst [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+:lang: en
+:authors: serge-sans-paille
+:summary: Pythran can also be used as a source-to-source transformation engine.
+          This post showcases some recent transformation on a high-level code.
+
+If you're curious or genuinely interested into how Pythran transforms your
+code, but not brave enough to dive into the generated C++ code, Pythran
+provides a compilation switch to dump the refined Python code, after
+optimization and before it gets translated to C++. Internally, this relies on
+the fact we have to backends: a C++ backend and a Python backend.
```
Comment (jeanlaroche) [inline content/2019-03-06-source-to-source-transformation.rst]: we have two backends

File: content/2019-03-06-source-to-source-transformation.rst [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    def fibo(n):
+        return (n if (n < 2) else (fibo((n - 1)) + fibo((n - 2))))
+    def test():
+        __builtin__.print(55)
+        return __builtin__.None
+
+What happened? Pythran analyzed the body of ``fibo`` and found out it was a
+pure function (no effect on global state nor arguments) called with a literal,
+so it performed aggressive constant propagation. It also computed variable
+scope and made any builtin explicit (``__builtin__.print``) and computed the
```
Comment (jeanlaroche) [inline content/2019-03-06-source-to-source-transformation.rst]: Can you define "variable scope" for me?

File: content/2019-03-06-source-to-source-transformation.rst [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+these operation!
+
+.. code-block:: shell
+
+    > pythran -P wsum.py
+    import numpy as __pythran_import_numpy
+    def wsum(v, w, x, y, z):
+        return __builtin__.sum(((v * 0.1), (w * 0.2), (x * 0.3), (y * 0.2), (z * 0.1)))
+
+Fascinating! (Yes, I'm auto-congratulating there). Pythran understood that a
```
Comment (jeanlaroche) [inline content/2019-03-06-source-to-source-transformation.rst]: "self-congratulating" is the more established way to say it! :)

File: content/2019-03-06-source-to-source-transformation.rst [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+.. code-block:: python
+
+    import numpy as __pythran_import_numpy
+    def wsum(v, w, x, y, z):
+        return __builtin__.sum(__pythran_import_numpy.array([(v * 0.1), (w * 0.2), (x * 0.3), (y * 0.2), (z * 0.1)]))
+
+Then it used the fact that sum can take any iterable as parameter to prune the
+call to ``np.array``. The nice thing with tuple of homogeneous type as
+parameter is that the C++ backend can use it to generate something equivalent
+to ``std::array<double, 5>``, avoiding a memory allocation.
```
Comment (jeanlaroche) [inline content/2019-03-06-source-to-source-transformation.rst]: Why  does std::array<double,5> avoid memory allocation?

**Comment by serge-sans-paille** [comment]
Thanks @jeanlaroche ! the second commit should answer tour questions ;-)

**Pull request merged**